File: Code.java
// Definition for singly-linked list:
// class ListNode<T> {
//   ListNode(T x) {
//     value = x;
//   }
//   T value;
//   ListNode<T> next;
// }
//
ListNode<Integer> mergeTwoLinkedLists(ListNode<Integer> l1, ListNode<Integer> l2) {
    ListNode<Integer> res = null;
    ListNode<Integer> temp = null;
    ListNode<Integer> prev = null;
    
    while(l1 != null || l2 != null)
    {           
        if(l2 == null || l1 != null && l1.value <= l2.value)
        {
            temp = new ListNode<Integer>(l1.value);
            l1 = l1.next;
        }
        else
        {
            temp = new ListNode<Integer>(l2.value);
            l2 = l2.next;
        }
        
        
        if(res == null)
            res = temp;
        else
            prev.next = temp;
        
        prev = temp;
    }
    
    return res;
}

File: Code.c
//
// Definition for singly-linked list:
// typedef struct list_##name {
//   type value;
//   struct list_##name *next;
// } list_##name;
//
// list_##name *alloc_list_##name(type v) {
//   list_##name *l = calloc(1, sizeof(list_##name));
//   l->value = v;
//   return l;
// }
//
list_integer * mergeTwoLinkedLists(list_integer * l1, list_integer * l2) {
    list_integer * res = NULL;
    
    if(l1 != NULL || l2 != NULL){
        if(l2 == NULL || l1 != NULL && l1->value <= l2->value){
            res = alloc_list_integer(l1->value);
            res->next = mergeTwoLinkedLists(l1->next, l2);
        }else{
            res = alloc_list_integer(l2->value);
            res->next = mergeTwoLinkedLists(l1, l2->next);
        }
    }

    
    return res;
}
